test/utils/checklist.spec.js

```diff
@@ -32,102 +40,236 @@ describe('utils', () => {
-          contentTypes: [Constants.AssetContentType.CODE],
-          uri: 'path/to/file',
-        })).toEqual({
+        expect(
+          ChecklistUtil.findAssetLanguagesAndDependencies({
+            type: Constants.AssetType.FILE,
+            contentTypes: [Constants.AssetContentType.CODE],
+            uri: 'path/to/file',
+          }),
+        ).toEqual({
           languages: [],
           dependencies: [],
         });
       });
     });
+
+    describe('findDataFiles', () => {
+      it('should return empty result when asset is null or undefined', () => {
+        expect(ChecklistUtil.findDataFiles(null)).toEqual({ datafiles: [] });
+        expect(ChecklistUtil.findDataFiles(undefined)).toEqual({ datafiles: [] });
+      });
+
+      it('should return empty result when asset is not a data file', () => {
+        expect(
+          ChecklistUtil.findDataFiles({
+            type: Constants.AssetType.FILE,
+            contentTypes: [Constants.AssetContentType.CODE],
+            uri: 'path/to/file.py',
+          }),
+        ).toEqual({ datafiles: [] });
+      });
+
+      it('should return data file name when asset is a data file', () => {
+        expect(
+          ChecklistUtil.findDataFiles({
+            type: Constants.AssetType.FILE,
+            contentTypes: [Constants.AssetContentType.DATA],
+            uri: 'path/to/file.csv',
+          }),
+        ).toEqual({ datafiles: ['file.csv'] });
+      });
+
+      it('should return data file names for nested data files', () => {
+        expect(
+          ChecklistUtil.findDataFiles({
+            type: Constants.AssetType.FOLDER,
+            contentTypes: [Constants.AssetContentType.DATA],
+            uri: 'path/to/folder',
+            children: [
+              {
+                type: Constants.AssetType.FILE,
+                contentTypes: [Constants.AssetContentType.DATA],
+                uri: 'path/to/folder/file1.csv',
+              },
+              {
+                type: Constants.AssetType.FILE,
+                contentTypes: [Constants.AssetContentType.DATA],
+                uri: 'path/to/folder/file2.csv',
+              },
+            ],
+          }),
+        ).toEqual({ datafiles: ['file1.csv', 'file2.csv'] });
+      });
+    });
+
+    describe('findEntryPointFiles', () => {
+      it('should return empty result when entryPoints is null or undefined', () => {
+        expect(ChecklistUtil.findEntryPointFiles(null)).toEqual({ entrypoints: [] });
+        expect(ChecklistUtil.findEntryPointFiles(undefined)).toEqual({ entrypoints: [] });
+      });
+
+      it('should return entry point file names for given entrypoint assets', () => {
+        expect(
+          ChecklistUtil.findEntryPointFiles([
+            {
+              uri: 'path/to/file1.py',
+            },
+            {
+              uri: 'path/to/file2.py',
+            },
+          ]),
+        ).toEqual({ entrypoints: ['file1.py', 'file2.py'] });
```

Comment:
This test is failing for me - it's getting back an empty array.  I actually would expect an empty array the way this test is set up - there are no attributes flagging these files as entry point files, so I would not expect them to be detected as such:

```
    - Expected  - 4
    + Received  + 1

      Object {
    -   "entrypoints": Array [
    -     "file1.py",
    -     "file2.py",
    -   ],
    +   "entrypoints": Array [],
```

Reply:
Oh ou, my bad! 🥲🥲 i copied the tests for somewhere and didn't change them accordingly here.. tbh, i wrote all this code in the fever hangover😂 so i knew there'd be such things, lemme double check all the code after changes.